### PR TITLE
String processing functions

### PR DESCRIPTION
**Pull request by Konstantsiy**
Implemented functions that reverses and capitalizes a string

### CODE REVIEW

**Comment by GoRunMaxim** [comment]
Good job! Couple notes for you: 

1. Probably you want to change README.md file: you could take a look on this web-site: https://www.makeareadme.com
2. I caught some errors with golangci-linter. You should add comment for exported functions and set your linter.

**Comment by GoRunMaxim** [comment]
Also, I suggest you to take a look on this webpage: https://chris.beams.io/posts/git-commit/

**Comment by GoRunMaxim** [comment]
1. Good, your commits looks much more better, but read this: https://chris.beams.io/posts/git-commit/
2. Did you run the tests? It failed, didn't it?

**Comment by Konstantsiy** [comment]
> 1. Good, your commits looks much more better, but read this: https://chris.beams.io/posts/git-commit/
> 2. Did you run the tests? It failed, didn't it?

One test was intentionally made incorrectly to show that the function only works with Latin and Cyrillic

**Comment by GoRunMaxim** [comment]
> > 1. Good, your commits looks much more better, but read this: https://chris.beams.io/posts/git-commit/
> > 2. Did you run the tests? It failed, didn't it?
> 
> One test was intentionally made incorrectly to show that the function only works with Latin and Cyrillic

Well, usually, we run all tests to see that they are working correctly. They shout not failed. Could we find the way for our function to work not only with Latin and Cyrillic?

**Comment by GoRunMaxim** [comment]
I don't really think that we need "task1" and "src" directories, do we? So you could remove them.

**Comment by Konstantsiy** [comment]
> > > 1. Good, your commits looks much more better, but read this: https://chris.beams.io/posts/git-commit/
> > > 2. Did you run the tests? It failed, didn't it?
> > 
> > 
> > One test was intentionally made incorrectly to show that the function only works with Latin and Cyrillic
> 
> Well, usually, we run all tests to see that they are working correctly. They shout not failed. Could we find the way for our function to work not only with Latin and Cyrillic?

**Comment by Konstantsiy** [comment]
> I don't really think that we need "task1" and "src" directories, do we? So you could remove them.



> > > 1. Good, your commits looks much more better, but read this: https://chris.beams.io/posts/git-commit/
> > > 2. Did you run the tests? It failed, didn't it?
> > 
> > 
> > One test was intentionally made incorrectly to show that the function only works with Latin and Cyrillic
> 
> Well, usually, we run all tests to see that they are working correctly. They shout not failed. Could we find the way for our function to work not only with Latin and Cyrillic?

I think I found a solution: using an internal library for unicode: https://golang.org/pkg/unicode

**Comment by GoRunMaxim** [comment]
> > I don't really think that we need "task1" and "src" directories, do we? So you could remove them.
> 
> > > > 1. Good, your commits looks much more better, but read this: https://chris.beams.io/posts/git-commit/
> > > > 2. Did you run the tests? It failed, didn't it?
> > > 
> > > 
> > > One test was intentionally made incorrectly to show that the function only works with Latin and Cyrillic
> > 
> > 
> > Well, usually, we run all tests to see that they are working correctly. They shout not failed. Could we find the way for our function to work not only with Latin and Cyrillic?
> 
> I think I found a solution: using an internal library for unicode: https://golang.org/pkg/unicode

You could try it and see, if it works with Belorussian language.

**Comment by Konstantsiy** [comment]
Now all tests work, not only with Latin and Cyrillic!

**Comment by GoRunMaxim** [comment]
Looks good! A couple notes: 

1.  "go mod tidy" may help you to clean up unused dependencies
2.  Probably commits should start with such words as "Add", "Fix". 
3.  What happened if we try to capitalize empty string?

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/Konstantsiy/go-internship
+
+go 1.13
```
Comment (papahmsolo) [inline go.mod]: Let's give this module a name that describes its purpose

File: README.md [hunk 2]
```diff
@@ -1 +1,7 @@
-# go-internship
\ No newline at end of file
+# Assignments for Golang internship
+## Task 1 - Reverse a string
+1. Implement a function that reverses a string. Cover your code with unit tests, use TDD
+   approach. Create a package for all the source files. Put your package in a module.
+   Publish your solution to a remote repo. Release the module under version 1.0
+2. Implement a function that capitalizes a string
+
```
Comment (papahmsolo) [inline README.md]: Let's describe this module from the user's perspective? What problems does it solve, what is it capable of?

File: main.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package main
```
Comment (papahmsolo) [inline main.go]: - Package comment
- This should be a library you suppose to import into your project, so let's rename the package and the file name if you want

File: main_test.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+			Expected: "",
+		},
+	}
+
+	for _, tc := range testTable {
+		result := CapitalizeString(tc.Source)
+
+		if result != tc.Expected {
+			t.Errorf("Incorrect result. Expect %s, got %s", tc.Expected, result)
+		}
```
Comment (papahmsolo) [inline main_test.go]: `%q` would look nicer for empty strings and leading/trailing spaces.

**Review by papahmsolo — CHANGES_REQUESTED** [review]
Nice job!

See comments below

**Comment by papahmsolo** [comment]
- Fix test
- Add Examples

**Review by papahmsolo — APPROVED** [review]

**Review by papahmsolo — APPROVED** [review]

**Pull request merged**